### PR TITLE
Fix feedback widget positioning + refactor to use popover

### PR DESCRIPTION
**Pull request by pettinarip**
image

## Description

To fix the FeedbackWidget positioning issue + fix the horizontal overflow #12035 + improve a11y aspects since we were using a wrong component (AlertDialog), this PR switches the base component to a Popover.

## Related Issue

#12035
fixes: https://github.com/ethereum/ethereum-org-website/issues/12184

### CODE REVIEW

File: src/components/FeedbackWidget.tsx [hunk 1]
```diff
@@ -29,48 +30,43 @@ type FixedDotProps = ButtonProps & {
...
-        transform: "scale(1.1)",
-        transition: "transform 0.2s ease-in-out",
-      }}
-      transition="transform 0.2s ease-in-out, width 0.25s ease-in-out,
+const FixedDot = forwardRef(
+  ({ children, bottomOffset, isExpanded, ...props }: FixedDotProps, ref) => {
+    const size = "3rem"
+    return (
+      <Button
+        ref={ref}
```
Comment (pettinarip) [inline src/components/FeedbackWidget.tsx:38]: Needed for the `PopoverTrigger` to work correctly.

File: src/components/FeedbackWidget.tsx [hunk 2]
```diff
@@ -29,48 +30,43 @@ type FixedDotProps = ButtonProps & {
...
+        position="absolute"
+        w={{ base: size, lg: isExpanded ? "15rem" : size }}
+        h={size}
+        bottom={{ base: `${bottomOffset + 1}rem`, lg: 4 }}
```
Comment (pettinarip) [inline src/components/FeedbackWidget.tsx:42]: Absolute position the button to the right inside the sticky container.

File: src/components/FeedbackWidget.tsx [hunk 3]
```diff
@@ -141,126 +137,129 @@ const FeedbackWidget = () => {
...
-              >
-                {t("feedback-widget-prompt")}
-              </Text>
-            </ScaleFade>
-          )}
-        </Box>
-      </FixedDot>
-
-      <AlertDialog
+    <Box position="sticky" bottom="0">
```
Comment (pettinarip) [inline src/components/FeedbackWidget.tsx:140]: Keep it at the bottom of the screen.

**Review by corwintines — CHANGES_REQUESTED** [review]
Nice, thanks @pettinarip 

Noticing that on mobile, there is an issue with this overflowing

Screenshot 2024-02-14 at 1 47 55 PM

Ive created this issue for this: https://github.com/ethereum/ethereum-org-website/issues/12184

It currently exists still, but would be good to fix in this I think.

**Comment by pettinarip** [comment]
@corwintines good catch! added a new variant to control the internal width of the popover.

**Comment by pettinarip** [comment]
@corwintines closing this in favor of #10997.

**Pull request closed without merging**